tests/integrations/src/agglayer_setup.rs

```diff
@@ -119,12 +125,13 @@ pub async fn start_agglayer(tmp_dir: &Path, l1: &L1Docker) -> (oneshot::Receiver
         }],
     });
 
-    let config_file = tmp_dir.join("config.toml");
+    let config_file = config_path.join("config.toml");
     let toml = toml::to_string_pretty(&config).unwrap();
     std::fs::write(&config_file, toml).unwrap();
 
+    let gracefull_shutdown_token = shutdown_token.clone();
     let handle = std::thread::spawn(move || {
-        _ = agglayer_node::main(config_file);
+        agglayer_node::main(config_file, Some(gracefull_shutdown_token)).unwrap();
```

Comment:
Could we print here some useful error info before panicking? Maybe we should refactor `start_agglayer` to return error and then bubble up